### PR TITLE
Fix replay_action not waiting for execute response

### PR DESCRIPTION
**Pull request by bduffany**

### CODE REVIEW

File: enterprise/tools/replay_action/replay_action.go [hunk 1]
```diff
@@ -394,14 +394,22 @@ func (r *Replayer) execute(ctx, srcCtx, targetCtx context.Context, action *repb.
...
 		DigestFunction:  sourceExecutionRN.GetDigestFunction(),
 	}
+	executeErr := make(chan error, *n)
 	for i := 1; i <= *n; i++ {
 		i := i
 		r.executeGroup.Go(func() error {
-			execute(targetCtx, r.execClient, r.destBSClient, i, sourceExecutionRN, execReq)
-			return nil
+			err := execute(targetCtx, r.execClient, r.destBSClient, i, sourceExecutionRN, execReq)
+			executeErr <- err
```
Comment (iain-macdonald) [inline enterprise/tools/replay_action/replay_action.go:402]: do you want to do this if err is nil? sorry, not sure what happens if you send nil to a chan

Comment (bduffany) [reply enterprise/tools/replay_action/replay_action.go:402]: yeah, this is just collecting all of the return values. I'd call this 'results' or something but I feel like 'result' is too overloaded of a term in this context :thinking:

Comment (iain-macdonald) [reply enterprise/tools/replay_action/replay_action.go:402]: sg, may want to log these somewhere but up to you

**Review by iain-macdonald — APPROVED** [review]

**Review by vanja-p — APPROVED** [review]

File: enterprise/tools/replay_action/replay_action.go (hunk 1 above)
Comment (bduffany) [reply enterprise/tools/replay_action/replay_action.go:402]: Done (we were logging only the last error before, but now we're logging every error)

**Pull request merged**